src/chunk.c

```diff
@@ -1134,13 +1135,37 @@ append_chunk_oid(ChunkScanCtx *scanctx, Chunk *chunk)
 		LockRelationOid(chunk->table_id, scanctx->lockmode);
 
 	scanctx->data = lappend_oid(scanctx->data, chunk->table_id);
+
 	return CHUNK_PROCESSED;
 }
 
-List *
-ts_chunk_find_all_oids(Hyperspace *hs, List *dimension_vecs, LOCKMODE lockmode)
+static ChunkResult
+append_chunk(ChunkScanCtx *scanctx, Chunk *chunk)
+{
+	Chunk **chunks = scanctx->data;
+
+	if (!chunk_is_complete(scanctx, chunk))
+		return CHUNK_IGNORED;
+
+	/* Fill in the rest of the chunk's data from the chunk table */
+	chunk_fill_stub(chunk, false);
+
+	if (scanctx->lockmode != NoLock)
+		LockRelationOid(chunk->table_id, scanctx->lockmode);
+
+	if (NULL == chunks && scanctx->num_complete_chunks > 0)
+		scanctx->data = chunks = palloc(sizeof(Chunk *) * scanctx->num_complete_chunks);
+
+	if (scanctx->num_complete_chunks > 0)
```

Comment:
I probably misunderstood a logic here, but does it makes sense to rely on `num_complete_chunks` counter here, since there are a check to ignore all non complete chunks at the beginning of the function?

Reply:
So, what this function does is to collect the "complete" chunks (those that have slices in all dimensions) after a scan. During the scan, we increment `num_complete_chunks` whenever a chunk gets "completed", then we know how many chunks to collect from the hash table at the end. This simply uses the `num_complete_chunks` as an index into the final array.